### PR TITLE
fix build showing success on failing project (windows)

### PR DESCRIPTION
**Pull request by krwq**
ftw !$?

### CODE REVIEW

**Comment by krwq** [comment]
lol csc.exe crashed (will report on their website).

**Comment by krwq** [comment]
@dotnet-bot test this please

File: scripts/compile.ps1 [hunk 1]
```diff
@@ -47,6 +52,11 @@ Download it from https://www.cmake.org
...
     header "Restoring packages"
     dotnet restore "$RepoRoot" --quiet --runtime "osx.10.10-x64" --runtime "ubuntu.14.04-x64" --runtime "win7-x64" --no-cache
+    if (!$?) {
+        Write-Host "Command failed: " dotnet restore "$RepoRoot" --quiet --runtime "osx.10.10-x64" --runtime "ubuntu.14.04-x64" --runtime "win7-x64" --no-cache
+        Exit 1
+    }
+
 
     header "Building corehost"
     pushd "$RepoRoot\src\corehost"
```
Comment (Sridhar-MS) [inline scripts/compile.ps1]: Let's add a check for core host build failure too.

Comment (krwq) [inline scripts/compile.ps1]: This is just pushd, it's not actually doing anything except changing cwd

Comment (krwq) [inline scripts/compile.ps1]: done

**Comment by Sridhar-MS** [comment]
:shipit: after the PR CI succeeds.

**Comment by krwq** [comment]
Note that with this change build stops immediately after the first error and will not continue building until error is fixed - I thought this is better since it is gonna push us more into fixing build issues.

**Comment by krwq** [comment]
Windows is passing, merging

**Pull request merged**